### PR TITLE
BaseComponent: Introduce generateId()

### LINKED ISSUE #97: BaseComponent: Introduce `generateId()`

### PR DESCRIPTION
**Pull request by awinogradov**
Close #97

### CODE REVIEW

File: lib/BaseComponent.js [hunk 1]
```diff
@@ -65,6 +67,13 @@ export default inherit(Component, {
 
     getChildContext() {
         if(!this.elem) return { bemBlock : this.block };
+    },
+
+    generateId(key = '__uniqId') {
+        this.__uniqId = this.__uniqId || {};
+        return this.__uniqId[key]
+            ? this.__uniqId[key]
```
Comment (veged) [inline lib/BaseComponent.js]: that's against current code style practice https://github.com/bem/bem-react-core/blob/master/lib/decls.js#L30 ;-) let's add lint check

File: lib/BaseComponent.js [hunk 2]
```diff
@@ -65,6 +67,13 @@ export default inherit(Component, {
 
     getChildContext() {
         if(!this.elem) return { bemBlock : this.block };
+    },
+
+    generateId(key = '__uniqId') {
+        this.__uniqId = this.__uniqId || {};
```
Comment (veged) [inline lib/BaseComponent.js]:
```js
this.__uniqId || (this.__uniqId = {});
```

File: lib/BaseComponent.js [hunk 3]
```diff
@@ -65,6 +67,13 @@ export default inherit(Component, {
...
+        return this.__uniqId[key]
+            ? this.__uniqId[key]
+            : (this.__uniqId[key] = `uniq$${++uniqCount}`);
```
Comment (veged) [inline lib/BaseComponent.js]: maybe
```js
`${key}${++uniqCount}`
```

File: __tests__/uniqId.spec.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import MyBlock from 'b:MyBlock';
+
+it('Should generate id', () => {
+    expect(shallow(<MyBlock/>).prop('id'))
+        .toMatch(/uniq\$/);
```
Comment (veged) [inline __tests__/uniqId.spec.js]: I suggest generation of two nodes inside content (i.e. for input/label) and check equality of attributes (which are based on `generateId()` from same context)

File: lib/BaseComponent.js (hunk 2 above)
Comment (awinogradov) [inline lib/BaseComponent.js]: Doesn't matter here ;)

Comment (veged) [inline lib/BaseComponent.js]: just the matter of code style ;-)

**Pull request merged**